prov/efa/src/efa_user_info.c

```diff
@@ -521,27 +521,38 @@ int efa_getinfo(uint32_t version, const char *node,
 		shm_info_initialized = true;
 	}
 
-	if (hints && hints->ep_attr && hints->ep_attr->type == FI_EP_DGRAM)
-		return efa_user_info_get_dgram(version, node, service, flags, hints, info);
+	if (hints && hints->ep_attr && hints->ep_attr->type == FI_EP_DGRAM) {
+		err = efa_user_info_get_dgram(version, node, service, flags, hints, info);
+		if (err)
+			goto error;
+		else
```

Comment:
It seems we do not need `else` and we can `return 0`.

Reply:
That's mostly just style right? Do you want it changed or is it okay as is?

Reply:
I will prefer the change if it is not too much trouble.

Reply:
Done